src/components/auth/SignUpScreen.js

```diff
@@ -1,290 +1,303 @@
+                    onChangeText={(value) => this.onChangeText("name", value)}
+                  />
+                </Item>
+                <Item style={styles.itemStyle}>
+                  <Input
+                    placeholder={i18n.t("Email")}
+                    placeholderTextColor={COLORS.forestgreen}
+                    returnKeyType="next"
+                    autoCapitalize="none"
+                    autoCorrect={false}
+                    keyboardType={"email-address"}
+                    ref="SecondInput"
+                    onSubmitEditing={(event) => {
+                      this.refs.ThirdInput._root.focus();
+                    }}
+                    onChangeText={(value) => this.onChangeText("email", value)}
+                  />
+                </Item>
+                <Item style={styles.itemStyle}>
+                  <Input
+                    placeholder={i18n.t("Password")}
+                    placeholderTextColor={COLORS.forestgreen}
+                    returnKeyType="next"
+                    autoCapitalize="none"
+                    autoCorrect={false}
+                    secureTextEntry={this.state.hidePassword}
+                    ref="ThirdInput"
+                    onSubmitEditing={(event) => {
+                      this.refs.FourthInput._root.focus();
+                    }}
+                    onChangeText={(value) =>
+                      this.onChangeText("password", value)
+                    }
+                  />
+                  <Ionicons
+                    style={
+                      this.state.hidePassword == false
+                        ? styles.showIcon
+                        : styles.hideIcon
+                    }
+                    name="ios-eye"
+                    onPress={() => this.handleHidePassword()}
+                  />
+                </Item>
+                <Item style={styles.itemStyle}>
+                  <Input
+                    placeholder={i18n.t("Confirm Password")}
+                    placeholderTextColor={COLORS.forestgreen}
+                    returnKeyType="go"
+                    autoCapitalize="none"
+                    autoCorrect={false}
+                    secureTextEntry={this.state.hidePassword}
+                    ref="FourthInput"
+                    onChangeText={(value) =>
+                      this.onChangeText("password_confirmation", value)
+                    }
+                  />
+                </Item>
+
+                {/* SignUp Button */}
+                <TouchableOpacity
+                  onPress={() => this.handleSignUp()}
+                  disabled={this.state.isLoading}
+                  style={styles.submitButton}>
+                  <Text style={styles.submitLabel}>{i18n.t("Sign Up")}</Text>
+                </TouchableOpacity>
+
+                {/* Loading ActivityIndicator */}
+                {this.state.isLoading && (
+                  <View>
+                    <ActivityIndicator
```

Comment:
Take a look at how Jules does this in REF-8. Let's place the activityIndicator inside of the button replacing the text when loading